### ISSUE TITLE
Get dirty fields from Form Hooks

### ISSUE THREAD

**Issue by DaniTulp** [tinacms/tinacms#990]
## Summary
Right now on Form Hooks we can check if the form is dirty. What I would like to be able to do is to check which fields are dirty. 

### Basic example
On the Form Hooks we have a form property and it contains the `fields` array. I would propose to add a property like `isDirty: false` or the inverse `valid: true`. This way the developer can check if a field is valid and do logic based on that.

### Motivation
Let's say I have a YoutubeField and based on if the link is valid I render a videoplayer, if it's not valid It should not render the player. Right now I would have to validate it in two places, once for Tina and once for rendering it. It would be nice to have one place to do it in.

**Comment by ncphillips** [comment]
Great request!

The `Form` interface doesn't support this directly but you can by-pass it to get this information from the `fina-form` instance. For example:

```
form.finalForm.getFieldState('video').valid
```

Checkout the [FieldState](https://final-form.org/docs/final-form/types/FieldState) docs for more information.


As an aside: `dirty` and `valid` are actually unrelated concepts. The `dirty` flag means that the field has been changed, the opposite is `pristine`.

**Comment by DaniTulp** [comment]
Thanks a lot @ncphillips! This works great and thanks for clarifying the concept dirty vs valid vs pristine.

**Comment by ncphillips** [comment]
Thank you for the idea!

@all-contributors please add @DaniTulp for ideas

**Comment by DaniTulp** [comment]
I actually ran into some problems I see now, due to hot-reloading they weren't obvious at first. When I use `form.finalForm.getFieldState('video').valid` the first render it is always undefined, I think it's probably not registered yet? I looked through the finalForm object some more and saw a property called `dirtyFields` so I dug some more and am currently using this solution `form.finalForm.getState().dirtyFields.video` which works great.
